tests/test_matrix_utils.py

```
@@ -0,0 +1,312 @@
+    assert credentials is None
+
+
+@patch("biblebot.auth.tempfile.NamedTemporaryFile")
+@patch("biblebot.auth.os.replace")
+def test_save_credentials(mock_replace, mock_temp_file):
+    """Test credentials saving with atomic write."""
+    from biblebot.auth import Credentials
+
+    # Mock temporary file with proper fileno and fsync
+    mock_temp = MagicMock()
+    mock_temp.name = "/tmp/temp_file"
```

Ruff (0.12.2)

Probable insecure usage of temporary file or directory: "/tmp/temp_file"

(S108)